Description: Multiple overlapping QML MouseAreas gave incorrect events
 Multiple overlapping QML MouseAreas gave incorrect events. This patch 
 prevents the issue.

Author: Daniel d'Andrada <daniel.dandrada@canonical.com>
Origin: Canonical
Forwarded: https://codereview.qt-project.org/#change,43270
Last-Update: 2012-12-15

---
--- qtdeclarative-5.0-release~beta+20120831.orig/src/quick/items/qquickwindow.cpp
+++ qtdeclarative-5.0-release~beta+20120831/src/quick/items/qquickwindow.cpp
@@ -430,8 +430,18 @@ bool QQuickWindowPrivate::translateTouch
             if (!item->contains(pos))
                 break;
 
-            bool doubleClick = event->timestamp() - touchMousePressTimestamp
-                            < static_cast<ulong>(qApp->styleHints()->mouseDoubleClickInterval());
+            bool doubleClick;
+            {
+                ulong timeBetweenPresses = event->timestamp() - touchMousePressTimestamp;
+                if (timeBetweenPresses != 0) {
+                    ulong doubleClickInterval = static_cast<ulong>(qApp->styleHints()->mouseDoubleClickInterval());
+                    doubleClick = timeBetweenPresses < doubleClickInterval;
+                } else {
+                    // we are still dealing with the same event that generated
+                    // the previous press timestamp
+                    doubleClick = false;
+                }
+            }
             touchMousePressTimestamp = event->timestamp();
             // Store the id already here and restore it to -1 if the event does not get
             // accepted. Cannot defer setting the new value because otherwise if the event
